cmd/cosign/cli/initialize/init.go

```diff
@@ -40,6 +40,18 @@ func DoInitializeWithRootChecksum(ctx context.Context, root, mirror, rootChecksu
 	return doInitialize(ctx, root, mirror, rootChecksum, false)
 }
 
+func DoInitializeStaging(ctx context.Context) error {
+	stagingRootFile, err := os.CreateTemp("", "staging-root")
+	if err != nil {
+		return err
+	}
+	defer stagingRootFile.Close()
+	if _, err := stagingRootFile.Write(tuf.StagingRoot()); err != nil {
```

Comment:
Writing the whole file out in a temp file is a little roundabout. You could just set `opts.Root = tuf.StagingRoot()` down after line 76 if a staging flag is set.